Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Slate/HandInteractionPanZoom.cs

```diff
@@ -260,79 +285,69 @@ private void UpdateIdle()
                 }
                 else
                 {
-                    if (velocityActive == true)
-                    {
-                        totalUVOffset = new Vector2(totalUVOffset.x * velocityDampingX, totalUVOffset.y * velocityDampingY);
-                        FirePanning(0);
-                    }
+                    totalUVOffset = new Vector2(totalUVOffset.x * momentumHorizontal, totalUVOffset.y * momentumVertical);
+                    FirePanning(0);
                 }
             }
         }
         private void UpdateUVMapping()
-        {
-            if (velocityActive)
+        {        
+            Vector2 tiling = currentMaterial != null ? currentMaterial.mainTextureScale : new Vector2(1.0f, 1.0f);
+            List<Vector2> uvs = new List<Vector2>();
+            List<Vector2> uvsOrig = new List<Vector2>();
+            Vector2 uvTestValue;
+            mesh.GetUVs(0, uvs);
+            uvsOrig.AddRange(uvs);
+            float scaleUVDelta = 0.0f;
+            Vector2 scaleUVCentroid = Vector2.zero;
+            float currentContactRatio = 0.0f;
+
+            if (scaleActive)
             {
-                bool oobX = false;
-                bool oobY = false;
-                Vector2 tiling = new Vector2(1.0f, 1.0f);
-                List<Vector2> uvs = new List<Vector2>();
-                List<Vector2> uvsOrig = new List<Vector2>();
-                mesh.GetUVs(0, uvs);
-                uvsOrig.AddRange(uvs);
-                float scaleUVDelta = 0.0f;
-                Vector2 scaleUVCentroid = Vector2.zero;
-                float currentContactRatio = 0.0f;
+                scaleUVCentroid = GetDisplayedUVCentroid();
+                currentContactRatio = GetUVScaleFromTouches();
+                scaleUVDelta = currentContactRatio / previousContactRatio;
+                previousContactRatio = currentContactRatio;
 
-                if (scaleActive)
+                for (int i = 0; i < uvs.Count; ++i)
                 {
-                    //scaleUVCentroid = GetScaleUVCentroid();
-                    scaleUVCentroid = GetDisplayedUVCentroid();
-                    currentContactRatio = GetUVScaleFromTouches();
-
-                    //todo fix this- totalUVScale resets to 1 when second hand touches.
-
-                    //hint: totalUVOffset works correctly- make scaleUVDelta work the same way
-                    scaleUVDelta = currentContactRatio / previousContactRatio;
-                    previousContactRatio = currentContactRatio;
+                    //this is where zoom is applied if Active
+                    uvs[i] = ((uvs[i] - scaleUVCentroid) / scaleUVDelta) + scaleUVCentroid;
                 }
+            }
 
+            //test for pan limits
+            Vector2 uvDelta = new Vector2(totalUVOffset.x, -totalUVOffset.y);
+            if (unlimitedPan == false)
```

Comment:
```suggestion
            if (!unlimitedPan)
```

Reply:
Nit: Usually you don't see boolean types being compared like:

if (unlimitedPan == true)

Usually it's just

if (unlimitedPan)